### PR TITLE
doc: add formatted output user manual

### PR DESCRIPTION
**Pull request by FeynmanZhou**
Add formatted output user manual. Most of the content is replicated from the [spec](https://github.com/oras-project/oras/blob/main/docs/proposals/formatted-output.md#proposal-and-desired-user-experience) because those examples are also helpful to end users. 

Preview link: https://deploy-preview-356--oras-project.netlify.app/docs/how_to_guides/format_output

### CODE REVIEW

File: versioned_docs/version-1.2/how_to_guides/format_output.mdx [hunk 1]
```diff
@@ -0,0 +1,304 @@
...
+```
+
+View the content of the generated manifest within specified `sample-manifest` file. The output should be compact JSON data:
+
+```console
+$ cat sample-manifest
+{"schemaVersion":2,"mediaType":"application/vnd.oci.image.manifest.v1+json","artifactType":"application/vnd.unknown.artifact.v1","config":{"mediaType":"application/vnd.oci.empty.v1+json","digest":"sha256:44136fa355b3678a1146ad16f7e8649e94fb4fc21fe77e8310c060f61caaff8a","size":2,"data":"e30="},"layers":[{"mediaType":"application/vnd.oci.image.layer.v1.tar","digest":"sha256:6cb759c4296e67e35b0367f3c0f51dfdb776a0c99a45f39d0476e43d82696d65","size":14477,"annotations":{"org.opencontainers.image.title":"sbom.spdx"}},{"mediaType":"application/vnd.oci.image.layer.v1.tar","digest":"sha256:54c0e84503c8790e03afe34bfc05a5ce45c933430cfd9c5f8a99d2c89f1f1b69","size":6639,"annotations":{"org.opencontainers.image.title":"scan-test-verify-image.json"}}],"annotations":{"org.opencontainers.image.created":"2023-12-15T09:41:54Z"}}
+```
+
+> [!NOTE]
```
Comment (sajayantony) [inline versioned_docs/version-1.2/how_to_guides/format_output.mdx:46]: something might be off here - 
image

I think the other notes also need to be fixed.

**Review by sajayantony — CHANGES_REQUESTED** [review]

File: versioned_docs/version-1.2/how_to_guides/format_output.mdx [hunk 2]
```diff
@@ -0,0 +1,312 @@
+---
+title: Formatted output
+sidebar_position: 100
+---
+
+ORAS allows to use `--output` to output a file or directory in the filesystem. To enable users to format the metadata output of ORAS commands and compute figures based on the formatted output, these two options are proposed as follows:
```
Comment (TerryHowe) [inline versioned_docs/version-1.2/how_to_guides/format_output.mdx:6]:
```suggestion
The ORAS command allows users to format the metadata output and perform computations on data based on the formatted output. The ORAS command also allows the `--output` option to write to a file or directory.
```

File: versioned_docs/version-1.2/how_to_guides/format_output.mdx [hunk 3]
```diff
@@ -0,0 +1,312 @@
...
+
+- Use `--format <DATA_FORMAT>` to format metadata output of ORAS commands into different formats including prettified JSON, tree, table view, and Go template, i.e. `--format json|tree|table|go-template=GO_TEMPLATE`. It supports computing figures within the template using [Sprig](http://masterminds.github.io/sprig/) functions. This is the primary and recommended usage.
```
Comment (TerryHowe) [inline versioned_docs/version-1.2/how_to_guides/format_output.mdx:8]:
```suggestion
- Use `--format <DATA_FORMAT>` to format metadata output of ORAS commands into different formats including prettified JSON, tree, table view, and Go template, i.e. `--format json|tree|table|go-template`. It supports computing figures within the template using [Sprig](http://masterminds.github.io/sprig/) functions. This is the primary and recommended usage.
```

File: versioned_docs/version-1.2/how_to_guides/format_output.mdx [hunk 4]
```diff
@@ -0,0 +1,312 @@
...
+- Use `--template GO_TEMPLATE` to compute and manipulate the output data using Go template based on the chosen data format. To avoid ambiguity, this flag can only be used along with `--format go-template`.
+
+Note that both `--format` and `--template` are marked as "Experimental" in its first iteration v1.2.0.
```
Comment (TerryHowe) [inline versioned_docs/version-1.2/how_to_guides/format_output.mdx:11]:
```suggestion
:::note

Both `--format` and `--template` are marked as "Experimental" in its first iteration v1.2.0.

:::
```

File: versioned_docs/version-1.2/how_to_guides/format_output.mdx [hunk 5]
```diff
@@ -0,0 +1,312 @@
...
+
+## How to format output in ORAS commands
+
+ORAS allows to use the `--format` flag to format metadata output into structured data (e.g. JSON) and optionally use the `--template` with the [Go template](https://pkg.go.dev/text/template) to manipulate the output data. See sample usages in several commands below.
```
Comment (TerryHowe) [inline versioned_docs/version-1.2/how_to_guides/format_output.mdx:15]:
```suggestion
Use the `--format` flag to format metadata output into structured data (e.g. JSON) and optionally use the `--template` option with a [Go template](https://pkg.go.dev/text/template) to manipulate the output data. See sample usages in several commands below.
```

File: versioned_docs/version-1.2/how_to_guides/format_output.mdx [hunk 6]
```diff
@@ -0,0 +1,312 @@
...
+View the content of the generated manifest within specified `sample-manifest` file. The output should be compact JSON data:
+
+```console
+$ cat sample-manifest
+{"schemaVersion":2,"mediaType":"application/vnd.oci.image.manifest.v1+json","artifactType":"application/vnd.unknown.artifact.v1","config":{"mediaType":"application/vnd.oci.empty.v1+json","digest":"sha256:44136fa355b3678a1146ad16f7e8649e94fb4fc21fe77e8310c060f61caaff8a","size":2,"data":"e30="},"layers":[{"mediaType":"application/vnd.oci.image.layer.v1.tar","digest":"sha256:6cb759c4296e67e35b0367f3c0f51dfdb776a0c99a45f39d0476e43d82696d65","size":14477,"annotations":{"org.opencontainers.image.title":"sbom.spdx"}},{"mediaType":"application/vnd.oci.image.layer.v1.tar","digest":"sha256:54c0e84503c8790e03afe34bfc05a5ce45c933430cfd9c5f8a99d2c89f1f1b69","size":6639,"annotations":{"org.opencontainers.image.title":"scan-test-verify-image.json"}}],"annotations":{"org.opencontainers.image.created":"2023-12-15T09:41:54Z"}}
+```
+
+:::note
+
+`--output -` and `--format` can not be used at the same time due to conflicts.
```
Comment (TerryHowe) [inline versioned_docs/version-1.2/how_to_guides/format_output.mdx:48]:
```suggestion
The `--output -` option for standard output cannot be used with the `--format` option due to conflicts.
```

**Review by TerryHowe — CHANGES_REQUESTED** [review]

**Comment by TerryHowe** [comment]
This was originally written as a requirements document and just needs some updates to change the tone.

File: versioned_docs/version-1.2/how_to_guides/format_output.mdx (hunk 2 above)
Comment (FeynmanZhou) [reply versioned_docs/version-1.2/how_to_guides/format_output.mdx]: Updated

File: versioned_docs/version-1.2/how_to_guides/format_output.mdx (hunk 3 above)
Comment (FeynmanZhou) [reply versioned_docs/version-1.2/how_to_guides/format_output.mdx]: Updated

File: versioned_docs/version-1.2/how_to_guides/format_output.mdx (hunk 4 above)
Comment (FeynmanZhou) [reply versioned_docs/version-1.2/how_to_guides/format_output.mdx]: Updated

File: versioned_docs/version-1.2/how_to_guides/format_output.mdx (hunk 5 above)
Comment (FeynmanZhou) [reply versioned_docs/version-1.2/how_to_guides/format_output.mdx]: Updated

File: versioned_docs/version-1.2/how_to_guides/format_output.mdx (hunk 6 above)
Comment (FeynmanZhou) [reply versioned_docs/version-1.2/how_to_guides/format_output.mdx]: Accepted and updated

**Pull request merged**